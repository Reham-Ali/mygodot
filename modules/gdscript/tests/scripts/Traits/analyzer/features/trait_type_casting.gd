trait Shearable:
    pass

trait Milkable:
    pass

class FluffyCow:
    uses Shearable, Milkable

class FluffyBull:
    uses Shearable

class Bird:
    pass

func test():
    var my_animals : Array = []
    my_animals.append(FluffyCow.new())
    my_animals.append(FluffyBull.new())
    my_animals.append(Bird.new())
    for animal in my_animals:
        var cast1 = animal as Shearable
        var cast2 = animal as Milkable
        print(cast1 != null)
        print(cast2 != null)

    print("ok")
