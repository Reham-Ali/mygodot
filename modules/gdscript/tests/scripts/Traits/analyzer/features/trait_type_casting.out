GDTEST_OK
>> WARNING
>> Line: 22
>> UNSAFE_CAST
>> Casting "Variant" to "Shearable" is unsafe.
>> WARNING
>> Line: 23
>> UNSAFE_CAST
>> Casting "Variant" to "Milkable" is unsafe.
true
true
true
false
false
false
ok
